### DISCUSSION TITLE
Importing an export fails

### DISCUSSION THREAD

**Discussion by joezappie** [outline/outline#3245] [Self Hosting] [answered]
I'm self hosting outline using docker compose. I was trying to add a backup service to my compose script and I took an export of all collections via the outline Export tool before starting. I messed up something up and lost my volumes for Outline. I tried to use the import function after recreating fresh volumes but I just get back "Upload failed, try again?". Trying again does not work.

![image](https://user-images.githubusercontent.com/6005836/158290849-01617659-f880-48a0-bd0f-9a11264b8877.png)

Checking the log I get the following output:
```
{"name":"attachments.create","modelId":null,"label":"processor","level":"info","message":"backlinks processing attachments.create"}
{"name":"attachments.create","modelId":null,"label":"processor","level":"info","message":"debouncer processing attachments.create"}
{"name":"attachments.create","modelId":null,"label":"processor","level":"info","message":"imports processing attachments.create"}
{"name":"attachments.create","modelId":null,"label":"processor","level":"info","message":"exports processing attachments.create"}
{"name":"attachments.create","modelId":null,"label":"processor","level":"info","message":"notifications processing attachments.create"}
{"name":"attachments.create","modelId":null,"label":"processor","level":"info","message":"revisions processing attachments.create"}
{"name":"attachments.create","modelId":null,"label":"processor","level":"info","message":"slack processing attachments.create"}
```

No errors in the logs.

I'm self hosting S3 using Minio. Checking my bucket, logging in successfully uploaded the avatar for my account to my bucket so it appears thats working correctly. I do have a bunch of images that are included in my backup.

**To Reproduce**
No idea how to reproduce this

**Expected behavior**
It should import my previously exported collections/documents

**Outline (please complete the following information):**
- Install: Self Hosted (docker compose)
- Version: ca0c67c44273

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Brave (Chromium)

**Comment by tommoor** [comment, marked as answer]
Converted to a discussion because this relates specifically to self hosting. You should check your browser network logs for the error in question as the upload failed from the browser – the file is uploaded directly to the minio container.

**Reply by joezappie** [reply]
Checking the browser logs I do see this:

`Failed to load resource: the server responded with a status of 400 () outline-wiki:1`

outline-wiki is the name of my s3 minio container. Not sure what to make of that.

My bucket has some info in it so its seems its able to connect to it:

![image](https://user-images.githubusercontent.com/6005836/158293341-497fa72f-cfa9-4246-8964-c06b0ffedc93.png)

I did extract my export and copy 1 collection out that doesn't contain images and rezipped it. Trying to upload that it does get further but errors on saying "Collection must exist". I did recreate an empty collection but I'm getting the same error. 

![image](https://user-images.githubusercontent.com/6005836/158293584-6422c619-4d42-46ab-b953-61a37854d78b.png)

![image](https://user-images.githubusercontent.com/6005836/158293682-477f078d-a2d8-4497-b7b9-b8dec305b740.png)

![image](https://user-images.githubusercontent.com/6005836/158293716-e47d7908-4465-4e68-a004-bb04958c43ac.png)

**Reply by joezappie** [reply]
I also tried making a new document and putting an image in it. It can upload just fine to my Minio server.

I think my first issue has to do with the file size as its 56mb. Copying out a collection that only has a few images (61kb) I get the 2nd error as before: "Failed • Collection must exist for document Software Docs".

**Reply by joezappie** [reply]
I found out that it starts to fail to upload after 25mb. I'm using nginx-proxy-manager and hosting minio/outline behind a reverse proxy. I thought that was limiting the file upload size, but from what I can tell the default is 1mb and nginx-proxy-manager defaults the body size to 2000mb uploads.

So seems like my original issue might be something external. One of my collections accounts for 54mbs. Deleting that from the archive and uploading it worked. 

Uploading a 30mb file directly through the minio interface works fine so I'm not sure what the difference is there.

**Reply by tommoor** [reply]
Maybe it was the `AWS_S3_UPLOAD_MAX_SIZE` environment variable – the example value is around 25Mb.

**Reply by joezappie** [reply]
Yup, that did it! I didn't even think to check my environment variables. Thanks for the help

**Reply by joezappie** [reply]
This might not be easily done, but maybe it would be good to have a separate max file size for admin imports or no limit at all. Assuming that the import/export feature is only an admin feature there's less risk to allowing them to upload a big backup. Having a separate 10-25mb limit for standard users to stop them from uploading 8K images off their phone is a different use case. Obviously the work around is to just change that number if you need to do an import, but in my case none of my other admins will have server access to change the docker compose file.

Let me know if you think that's worth while and I can make a feature request issue.

**Reply by tommoor** [reply]
This value mainly exists for the cloud hosted solution, so there we're not interested in separate limits depending on the user role. However I think there is an argument for a separate limit specifically for bulk import – it has come up before.

**Comment by heyvaldemar** [comment]
hey guys, I wrote [a detailed guide](https://www.heyvaldemar.com/install-outline-and-keycloak-using-docker-compose/) after I was able to configure Outline with MinIO and Keycloak.

Hopefully, it will help others.
